app/src/ui/branches/branch-list-item.tsx

```diff
@@ -18,10 +20,61 @@ interface IBranchListItemProps {
 
   /** The characters in the branch name to highlight */
   readonly matches: IMatches
+
+  /** Specifies whether the branch is local */
+  readonly isLocal: boolean
+
+  readonly onRenameBranch?: (branchName: string) => void
+
+  readonly onDeleteBranch?: (branchName: string) => void
 }
 
 /** The branch component. */
 export class BranchListItem extends React.Component<IBranchListItemProps, {}> {
+  private onContextMenu = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.preventDefault()
+
+    /*
+      There are multiple instances in the application where a branch list item
+      is rendered. We only want to be able to rename or delete them on the
+      branch dropdown menu. Thus, other places simply won't provide these
+      methods, such as the merge and rebase logic.
+    */
+    if (
+      this.props.onRenameBranch === undefined &&
+      this.props.onDeleteBranch === undefined
+    ) {
+      return
+    }
+
+    // TODO: find out - does branch menu have "Rename..." because ... stand
+    // for branch name? If so is it needed here? (same for delete)
+    // Also, branch menu has & in front for not mac, what is that?
+    // Should it be disabled for current branch? - note from prev PR
```

Comment:
@niik This todo was questions I add after reviewing the code for the top menu in the build-default-ment.ts
![image](https://user-images.githubusercontent.com/75402236/105480347-a0944c80-5c73-11eb-9b9b-11e5c15a340c.png)

I went without the ... presuming it was indicating the branch name which is clear when right clicking  on it.. but is it some convention I should following in the context menu?

Also what is the & for?

Reply:
The `...` suffix is a UX convention which means that the action in question isn't just gonna happen when you click on the item, i.e. there'll be some form of confirmation or other action needed before said action actually happens.

Here's a good example where we conditionally add the ellipsis depending on whether the user has chosen to see a confirmation prompt when stashing changes or not

https://github.com/desktop/desktop/blob/42a5be758c2ff19ae6ea44721db3556b8c16ae7c/app/src/ui/changes/changes-list.tsx#L354-L359

The ampersand is a Windows specific thing which indicates that the following character is the [access key](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/bb226831(v=vs.85)) meaning that a user can press <kbd>Alt</kbd> and that key instead of clicking on the item. We typically only use access keys in the top level menu so you can go  ahead and remove it from the context menu.